### ISSUE TITLE
Suggestion: maks feathering to the inside

### ISSUE THREAD

**Issue by waxfingers** [aferrero2707/PhotoFlow#127]
Would be useful if masks had the option to feather to the inside, alone or together with fearing to the outside.

**Comment by aferrero2707** [comment]
Sorry, I don't understand what you mean by "feather to the inside/outside"...

**Comment by waxfingers** [comment]
Sorry Mr. it's my garbage explanation. Right now when you make a path's mask it can be feathered but always to the exterior of the shape, IMO would be useful if we could change that behaviour to be ONLY or ALSO to the interior of the path shape. In other words a way of inverting the path's feather direction or feather it in both directions. HDR masking also would benefit ,-)

**Comment by aferrero2707** [comment]
Ah, now I understand! This is exactly the pour pose of the curve associate to the path: to make a non linear feathering, including a ring-shaped one if you set both end points to 0 and an intermediate point to 1. There is also an "invert" checked box to quickly invert the mask. I can post a couple of screenshots a bit later if you want...

**Comment by waxfingers** [comment]
I'm not sure we're on the same page of the bibble; seriously though, this is what I meanto be able to do this (both feathering enabled), just (normal) to the outside, or just to the inside =)
![pf_mask_feather](https://cloud.githubusercontent.com/assets/13238340/20632696/5d3b526a-b337-11e6-902c-8b20f008dc2d.jpg)

**Comment by aferrero2707** [comment]
Here is an example of asymmetric double-feathering:

<img width="1280" alt="screen shot 2016-11-25 at 19 02 25" src="https://cloud.githubusercontent.com/assets/6229237/20632960/ceb0a4f4-b341-11e6-8950-4bad92910611.png">

**Comment by waxfingers** [comment]
> Here is an example of asymmetric double-feathering:

Ha ha ha you wanted to say that XD
That's a usefull donut, but still can that be done without masking the center? Cheers mate

**Comment by aferrero2707** [comment]
Like this?

<img width="1280" alt="screen shot 2016-11-25 at 19 53 13" src="https://cloud.githubusercontent.com/assets/6229237/20633594/d8893804-b348-11e6-9bc9-63b3ac23a8e4.png">

**Comment by waxfingers** [comment]
Kind of, but you've got to understand that the idea is to draw a shape and inverting it and being able to feather the mask from the wireframe itself to the other (current) direction, we need a beer, here's a nother grab  
![pf_totheinside](https://cloud.githubusercontent.com/assets/13238340/20634986/e38258de-b34e-11e6-81cc-f4d4a3b22b22.jpg)

**Comment by aferrero2707** [comment]
Now I get it... I'm a bit slow this evening ;-) 

I need to think about the problem... it might become difficult to implement in the case when the wireframe is convex in some places, like this:

<img width="309" alt="screen shot 2016-11-25 at 22 54 24" src="https://cloud.githubusercontent.com/assets/6229237/20635776/23f72b02-b362-11e6-902b-604a0bca3944.png">

How would the feathering look like in this case? It's not trivial...

**Comment by waxfingers** [comment]
You're right, in this case feathering you've inverted the mask, so (inverting) the fether or **falloff** to the _inside_ (not really a good choice of words), _to the opposite direction_ would yield that same mask but  with a falloff starting from the borders of the shape, what I call "wireframe" to the center where the cross is. Yes if feathered too much would have strange overlapping areas... but that is already happening in PF's masks as they are, don't think you should be concern about it =)

As an example, the advantage of being able to do such a thing is that imagine I want to darken everything but doggy's left eye, I could make a precise shape mask around the black of left eye's socket, invert it and then enable _inverted falloff_ that would go from the edge of drawn shape to its iris. Trees and sharp edged objects are a better example and this, I believe, would make things a bit  easier. Ticks splitting hairs { |./ }

**Comment by aferrero2707** [comment]
There are two other possibilities to obtain what you want:

* create a non-feathered shape and add a Gaussian blur layer
* use a radial gradient, which does exactly what you want but limited to a circular shape. In future I plan to add elliptical shapes as well as an easier way to move around the gradient and edit the fall off...

**Comment by waxfingers** [comment]
Good to know, thanks
